packages/ni.measurementlink.sessionmanagement.v1.client/src/ni/measurementlink/sessionmanagement/v1/client/_client.py

```diff
@@ -42,25 +44,19 @@ def __init__(
         grpc_channel: grpc.Channel | None = None,
         grpc_channel_pool: GrpcChannelPool | None = None,
     ) -> None:
-        """Initialize session management client.
-
-        Args:
-            discovery_client: An optional discovery client (recommended).
-
-            grpc_channel: An optional session management gRPC channel.
-
-            grpc_channel_pool: An optional gRPC channel pool (recommended).
-        """
-        self._initialization_lock = threading.Lock()
-        self._discovery_client = discovery_client
-        self._grpc_channel_pool = grpc_channel_pool
+        """Initialize a SessionManagementClient instance."""
+        super().__init__(
+            discovery_client=discovery_client,
+            grpc_channel=grpc_channel,
+            grpc_channel_pool=grpc_channel_pool,
+            service_interface_name=GRPC_SERVICE_INTERFACE_NAME,
+            service_class=GRPC_SERVICE_CLASS,
+            stub_class=session_management_service_pb2_grpc.SessionManagementServiceStub,
+        )
+        self._discovery_client: DiscoveryClient | None = discovery_client
+        self._grpc_channel_pool: GrpcChannelPool | None = grpc_channel_pool
         self._stub: session_management_service_pb2_grpc.SessionManagementServiceStub | None = None
 
-        if grpc_channel is not None:
-            self._stub = session_management_service_pb2_grpc.SessionManagementServiceStub(
-                grpc_channel
-            )
-
     def _get_stub(self) -> session_management_service_pb2_grpc.SessionManagementServiceStub:
```

Comment:
I would expect _client.py to remove _get_stub and instead use the _get_stub from the base class.